llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -4147,6 +4147,22 @@ AArch64TTIImpl::getArithmeticReductionCost(unsigned Opcode, VectorType *ValTy,
   switch (ISD) {
   default:
     break;
+  case ISD::FADD: {
+    if (MTy.isVector()) {
+      // FIXME: Consider cases where the number of vector elements is not power
+      // of 2.
+      const unsigned NElts = MTy.getVectorNumElements();
+      if (ValTy->getElementCount().getFixedValue() >= 2 && NElts >= 2 &&
```

Comment:
I'm not sure if this is right. What about scalable vectors? Do we lower to faddp using SVE as well?

Reply:
This is only for fixed length vectors. For scalable vectors, the  [control flow](https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp#L4102) is different.

I havent checked the case with SVE to be honest.